STM32 : get serial RX/TX active state

jeromecoutant opened the pull request:
### Description

We don't report any real issue yet, patch is proposed thanks to some code review.

In serial_tx_active and serial_rx_active functions,
we check the internal state value with

  HAL_UART_STATE_BUSY_TX           = 0x21U,
  HAL_UART_STATE_BUSY_RX           = 0x22U,

It seems that value can also be :
  HAL_UART_STATE_BUSY_TX_RX     = 0x23U,

### Pull request type

    [x] Fix
    [ ] Refactor
    [ ] New target
    [ ] Feature
    [ ] Breaking change

0xc0170 reviewed (changes requested):
commit msg is `STM32 serial RX/TX active patch`

Should the second paragraph contain the info from description? It's small change but to me that description made it clear what is changing here

0xc0170 commented:
/morph build

cmonr commented:
@jeromecoutant Looks like the astyle PR has caused this to need a rebase.

jeromecoutant committed:
STM32 serial RX/TX active patch

In serial_tx_active and serial_rx_active functions,
we check the internal state value with

HAL_UART_STATE_BUSY_TX = 0x21U,
HAL_UART_STATE_BUSY_RX = 0x22U,

It seems that value can also be :
HAL_UART_STATE_BUSY_TX_RX = 0x23U,

jeromecoutant commented:
Rebase done

cmonr commented:
/morph build

cmonr commented:
https://github.com/ARMmbed/mbed-os/pull/7361#issuecomment-401126809 This was from the first build. Ignore.
